ui/src/app/views/error-handler/component.ts

```diff
@@ -0,0 +1,27 @@
+// UI Component framing the overall app (title and nav).
+// Content is in other Components.
+
+import {ChangeDetectorRef, Component, OnInit} from '@angular/core';
+import 'clarity-icons';
+@Component({
+  selector: 'app-error-handler',
+  styleUrls: ['./component.css'],
+  templateUrl: './component.html'
+})
+export class ErrorHandlerComponent implements OnInit {
+  notifyFiveHundred = false;
+
+  constructor() {}
+
+  ngOnInit(): void {
+    window['handleFiveHundred'] = () => {
```

Comment:
This breaks compile-checking, right? Can we make this an explicit / compiled link? Maybe register an event handler and publish an event from the retry code, if a direct method call won't work?

Reply:
I'm coming very late to the party and focusing on a detail, but: setting a handler function on `window` seems like a "code smell". Why not provide an error notification service, and have the main app component display these errors as needed?

Reply:
Brian and I were chatting about this; he suggested a service to handle the communication. Something similar to http://jasonwatmore.com/post/2017/06/25/angular-2-4-alert-toaster-notifications .